src/python/pants/backend/java/dependency_inference/java_parser_launcher.py

```diff
@@ -86,22 +90,23 @@ async def build_processors(bash: BashBinary, javac: JavacBinary) -> JavaParserCo
         ProcessResult,
         Process(
             argv=[
-                bash.path,
-                javac.javac_wrapper_script,
+                *jdk_setup.args(bash, [f"{jdk_setup.java_home}/lib/tools.jar"]),
```

Comment:
Shouldn't we be invoking `javac` and not assuming existence of non-`java.` packages? I.e., what is someone uses a JVM that does not have `com.sun.tools.javac.Main` as an entrypoint?

Reply:
Nailgun is not aware of `javac` in particular, and requires the main class as the first argument to the server.